synapse/lib/stormlib/storm.py

```diff
@@ -13,6 +15,76 @@
 
 stormlogger = logging.getLogger('synapse.storm')
 
+class StormExecCmd(s_storm.Cmd):
+    '''
+    Execute text or an embedded query object as Storm in the current pipeline.
+
+    NOTE: It is recommended to avoid using this where possible to avoid potential
+    query injection risks. If you must use this, take care to ensure any values
+    being executed have been properly sanitized.
+
+    Examples:
+
+        // Add nodes using text in a variable
+        $query = '[ inet:fqdn=foo.com inet:fqdn=bar.net ]'
+        storm.exec $query
+
+        // Filter nodes in the pipeline using text in a variable
+        $filter = '-:asn=10'
+        inet:ipv4:asn
+        storm.exec $filter
+
+        // Pivot using an embedded query
+        $pivot = ${ -> inet:asn }
+        inet:ipv4:asn
+        storm.exec $pivot
+    '''
+    name = 'storm.exec'
+    def getArgParser(self):
+        pars = s_storm.Cmd.getArgParser(self)
+        pars.add_argument('query', help='The Storm to execute.')
+        return pars
+
+    async def execStormCmd(self, runt, genr):
+
+        if self.runtsafe:
+
+            text = await s_stormtypes.tostr(self.opts.query)
+            query = await runt.getStormQuery(text)
+
+            extra = await self.runt.snap.core.getLogExtra(text=text)
+            stormlogger.info(f'Executing storm query via storm.exec {{{text}}} as [{self.runt.user.name}]', extra=extra)
+
+            async with runt.getSubRuntime(query) as subr:
+                async for subp in subr.execute(genr=genr):
+                    yield subp
+
+        else:
+
+            item = None
+            async for item in genr:
+                break
+
+            text = await s_stormtypes.tostr(self.opts.query)
+            query = await runt.getStormQuery(text)
+
+            extra = await self.runt.snap.core.getLogExtra(text=text)
+            stormlogger.info(f'Executing storm query via storm.exec {{{text}}} as [{self.runt.user.name}]', extra=extra)
+
+            async with runt.getSubRuntime(query) as subr:
+                async for subp in subr.execute(genr=s_common.agen(item)):
+                    yield subp
+
+                async for item in genr:
+                    text = await s_stormtypes.tostr(self.opts.query)
+                    subr.query = await runt.getStormQuery(text)
```

Comment:
This may be bypassing the runtvar loading/detection?

Reply:
Good point, adjusted to re-initialize runtvars for each per-node execution